CMakeLists.txt

```diff
@@ -180,52 +183,62 @@ message(STATUS "")
-  # On windows copy the mimalloc redirection dll too.
-  target_link_libraries(mimalloc PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/bin/mimalloc-redirect.lib)
-  add_custom_command(TARGET mimalloc POST_BUILD
-    COMMAND "${CMAKE_COMMAND}" -E copy "${CMAKE_CURRENT_SOURCE_DIR}/bin/mimalloc-redirect.dll" $<TARGET_FILE_DIR:mimalloc>
-    COMMENT "Copy mimalloc-redirect.dll to output directory")
+if(MI_BUILD_SHARED)
+  message(STATUS "Shared library will be built")
+
+  add_library(mimalloc SHARED ${mi_sources})
+  set_target_properties(mimalloc PROPERTIES VERSION ${mi_version} OUTPUT_NAME ${mi_basename} )
+  target_compile_definitions(mimalloc PRIVATE ${mi_defines} MI_SHARED_LIB MI_SHARED_LIB_EXPORT)
+  target_compile_options(mimalloc PRIVATE ${mi_cflags})
+  target_link_libraries(mimalloc PUBLIC ${mi_libraries})
+  target_include_directories(mimalloc PUBLIC
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+      $<INSTALL_INTERFACE:${mi_install_dir}/include>
+  )
+  if(WIN32)
+    # On windows copy the mimalloc redirection dll too.
+    target_link_libraries(mimalloc PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/bin/mimalloc-redirect.lib)
+    add_custom_command(TARGET mimalloc POST_BUILD
+      COMMAND "${CMAKE_COMMAND}" -E copy "${CMAKE_CURRENT_SOURCE_DIR}/bin/mimalloc-redirect.dll" $<TARGET_FILE_DIR:mimalloc>
+      COMMENT "Copy mimalloc-redirect.dll to output directory")
+  endif()
+
+  install(TARGETS mimalloc EXPORT mimalloc DESTINATION ${mi_install_dir} LIBRARY)
+  install(EXPORT mimalloc DESTINATION ${mi_install_dir}/cmake)
 endif()
 
 # static library
-add_library(mimalloc-static STATIC ${mi_sources})
-target_compile_definitions(mimalloc-static PRIVATE ${mi_defines} MI_STATIC_LIB)
-target_compile_options(mimalloc-static PRIVATE ${mi_cflags})
-target_link_libraries(mimalloc-static PUBLIC ${mi_libraries})
-target_include_directories(mimalloc-static PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-    $<INSTALL_INTERFACE:${mi_install_dir}/include>
-)
-if(WIN32)
-  # When building both static and shared libraries on Windows, a static library should use a
-  # different output name to avoid the conflict with the import library of a shared one.
-  string(REPLACE "mimalloc" "mimalloc-static" mi_output_name ${mi_basename})
-  set_target_properties(mimalloc-static PROPERTIES OUTPUT_NAME ${mi_output_name})
-else()
-  set_target_properties(mimalloc-static PROPERTIES OUTPUT_NAME ${mi_basename})
+if (MI_BUILD_STATIC)
+  message(STATUS "Static library will be built")
+
+  add_library(mimalloc-static STATIC ${mi_sources})
+  target_compile_definitions(mimalloc-static PRIVATE ${mi_defines} MI_STATIC_LIB)
+  target_compile_options(mimalloc-static PRIVATE ${mi_cflags})
+  target_link_libraries(mimalloc-static PUBLIC ${mi_libraries})
+  target_include_directories(mimalloc-static PUBLIC
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+      $<INSTALL_INTERFACE:${mi_install_dir}/include>
+  )
+  if(WIN32 AND MI_BUILD_SHARED)
```

Comment:
Ah, on WIN32 the static library should always be `mimalloc-static` so the `AND MI_BUILD_SHARED` should not be there.